### DISCUSSION TITLE
Clear Text Within Algolia Search Textbox After Submitting

### DISCUSSION THREAD

**Discussion by frscott** [quarto-dev/quarto-cli#7643] [Q&A]
### Description

I selected the textbox option rather than the default option where the overlay appears after clicking the magnifying glass. I noticed after submitting my search, the textbox does not clear the text. Which is a different behavior than the default overlay option, is this on purpose or is there a way to change it to clear the text in the textbox after submitting?

````qmd
website:
  search:
    location: navbar
    type: textbox
````

**Comment by mcanouil** [comment]
The behaviour is the same.
The input remains as long as you stay in the page of the results.


https://github.com/quarto-dev/quarto-cli/assets/8896044/eb9e8d02-9bed-4a93-85d6-f2e552675917

**Reply by frscott** [reply]
Yes you are right so it clears the text when you are directed to a new page, but not when directed to an anchor tag. Is there a way to have the text be cleared when pointing to just an anchor tag at the end of a URL? Here is my [example](https://mtftablesandfigures.com/bx-by/drug-prevalence/) I am working on. Any tips?

I have the search set up so that it populates specific drug names. If I search "Adderall" it will link to https://mtftablesandfigures.com/bx-by/drug-prevalence/#drug=%22Adderall%22 so the #drug="drug-name" is all that is changing in my dashboard.

**Reply by mcanouil** [reply]
You could use JavaScript to target the form and clear it on submit/action.

**Reply by frscott** [reply]
I tried but I can't seem to reference the form. I have never handled forms in JavaScript before. Something like this isn't working. I am not sure where the error is.

```
document.forms[0].addEventListener("submit", function(event) {document.getElementById("autocomplete-0-input").value = "";});
```

**Reply by mcanouil** [reply]
I am not completely sure about the code required to do that, but why do you want to clear the search just after it? Even web search engine don't do that.

Note that the search is controlled already by JavaScript which you need to account for.

https://github.com/quarto-dev/quarto-cli/blob/main/src/resources/projects/website/search/quarto-search.js

**Reply by frscott** [reply]
Yes I found the quarto-search.js file and that is where it became too complicated for me to follow. I was trying to edit that .js file with very little success.

I personally do not mind it. My boss was asking if it was possible, and I thought it was because I noticed it clearing when it took me to another page. I can just tell him it probably is possible, but I do not have enough JavaScript knowledge to figure it out.
